staging/src/k8s.io/client-go/rest/request_test.go

```diff
@@ -488,6 +488,14 @@ func TestURLTemplate(t *testing.T) {
 			ExpectedFullURL:  "http://localhost/some/base/url/path/pre1/namespaces/namespaces/namespaces/namespaces/namespaces/namespaces/finalize",
 			ExpectedFinalURL: "http://localhost/%7Bprefix%7D",
 		},
+		{
+			// dynamic client with version path
+			// /apis/$NAMEDGROUPNAME/$RESOURCEVERSION/namespaces/$NAMESPACE/$RESOURCE/%NAME/$SUBRESOURCE
+			Request: NewRequest(nil, "DELETE", uri, "", ContentConfig{GroupVersion: &schema.GroupVersion{Group: "test"}}, Serializers{}, nil, nil, 0).
+				Prefix("/version"),
+			ExpectedFullURL:  "http://localhost/some/base/url/path/version",
+			ExpectedFinalURL: "http://localhost/version",
```

Comment:
I do not understand why we would want to do this.

Reply:
agree, a trailing /version segment does not collapse to /version

Reply:
That's correct. @lavalamp @liggitt how do you propose that we fix this? Shall I ignore the `some/base/url` and only return the `FinalURL` as `http://localhost/version` if `FullURL` is `http://localhost/version`?

Reply:
Why do we want to cut the base url and only want the version?